mesonbuild/dependencies/ui.py

```diff
@@ -493,3 +493,56 @@ def check_wxconfig(self):
                 pass
         WxDependency.wxconfig_found = False
         mlog.log('Found wx-config:', mlog.red('NO'))
+        
+class VulkanDependency(ExternalDependency):
+    def __init__(self, environment, kwargs):
+        super().__init__('vulkan', environment, None, kwargs)
+        
+        if DependencyMethods.SYSTEM in self.methods:
+            try:
+                self.vulkan_sdk = os.environ['VULKAN_SDK']
+                if not os.path.isabs(self.vulkan_sdk):
+                    raise DependencyException('VULKAN_SDK must be an absolute path.')
+                    
+                # TODO: this config might not work on some platforms, fix bugs as reported
+                # we at least have to detect other 64-bit platforms (e.g. armv8)
+                lib_name = 'vulkan'
+                if mesonlib.is_windows():
+                    lib_name = 'vulkan-1'
+                    
+                lib_dir = 'Lib32'
+                if detect_cpu({}) == 'x86_64':
+                    lib_dir = 'Lib'
+                    
+                self.type_name = 'vulkan_sdk'
+                self.is_found = True
+                self.compile_args.append('-I' + os.path.join(self.vulkan_sdk, 'Include'))
+                self.link_args.append('-L' + os.path.join(self.vulkan_sdk, lib_dir))
+                self.link_args.append('-l' + lib_name)
```

Comment:
Still not convinced that it is needed. A dynamic loader usually provides a substitute header (the way GLEW does for GL). It would manually reparse the vulkan spec and provide everything needed.
The only example i found is !(this)[https://github.com/MrVallentin/vkel], not sure if there are others atm, but i think they all usually work without the vulkan header (which would pull the global vulkan functions in what is not wanted when they should be loaded dynamically).

IMO, in the end it would probably make more sense to add the possibility for using only certain fields of a dependency and not implement it for every dependency (you can probably bring similar arguments for other apis) if it is really needed.